rust/util/mod.rs

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#[macro_export]
+macro_rules! enum_getter {
+    ($fn_name:ident, $enum_value:ident, $classname:ty) => {
+        pub fn $fn_name(self) -> $classname {
+            match self {
+                Self::$enum_value(x) => x,
+                _ => panic!(""),
+            }
+        }
+    };
+}
```

Comment:
Lets us replace repetitive mostly identical code with simple declaration. If token concatenation were possible in non-proc macros, this could have been even terser.